### ISSUE TITLE
Should an aside in main fail axe?

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#2651]
We keep getting feedback about failing this scenario:

```html
<main>
  <p>Lorem's ipsum</p>
  <aside>Some stuff</aside>
</main>
```

Axe-core goes by the ARIA standards on this, but lets do some actual testing to figure out if this creates problems.

**Comment by WilcoFiers** [comment]
Did a little testing. Here's my code:

```html
  <main>
    <header>Header</header>
    <footer>Footer</footer>
    <aside>Aside</aside>
  </main>
```

- Chrome sees `main` and `complementary`
- Safari sees `main`, `banner` and `complementary`
- Firefox sees `main` and `complementary`

Not sure why Safari also treats considers the `heading` to be a landmark. If I put the `footer` outside the `main` it does pick it up. That seems like a bug in Safari.

**Comment by scottaohara** [comment]
would agree that safari is incorrect here.

**Comment by WilcoFiers** [comment]
I'm leaning towards creating an exception for `aside` in this rule. The `landmark-unique` rule will still call it out if someone uses multiple `aside` elements in the page, except if they give it a unique name, which I think also addresses the concern here.

**Comment by padmavemulapati** [comment]
Verified with the latest axe-core develop branch code base(dated 12th jan 2021):
rule:`landmark-complementary-is-top-level` verified with the pass and fail criteria validated to allow `aside` inside `main`
Test snippet for fail:
`<nav><div role="complementary" id="target"></div></nav>
 ` <!--should return true if the complementary landmark is in main landmark-->`
  `<main><div role="complementary" id="target1"></div></main>`
  `<nav>`
    `<div role="complementary">`
     ` <p>This complementary landmark is in a main landmark</p>`
    `</div>`
    `</nav>`

![image](https://user-images.githubusercontent.com/47409579/104331156-c6538000-5514-11eb-8d0b-f68791fa5f98.png)

Test snippet for pass:
`<div role="main">`
      `<div role="complementary">`
       ` <p>This div has role complementary></p>`
      `</div>`
    `</div>`

![image](https://user-images.githubusercontent.com/47409579/104331475-1d595500-5515-11eb-8d53-5edeb1fbc213.png)
